inference-engine/src/transformations/src/transformations/op_conversions/fq_decomposition.cpp

```diff
@@ -0,0 +1,148 @@
+
+        size_t count_not_unit_axis = 0;
+        auto shape = node->get_shape();
+        if (ngraph::shape_size(shape) != 1) {
+            size_t not_unit_axis = 0;
+            for (size_t i = 0; i < shape.size(); i++) {
+                if (shape[i] > 1) {
+                    not_unit_axis = i;
+                    count_not_unit_axis++;
+                }
+            }
+            if (count_not_unit_axis > 1 || not_unit_axis > 1)
+                return true;
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Expression from specification:
+ * if x <= min(input_low, input_high):
+ *   output = output_low
+ * elif x > max(input_low, input_high):
+ *   output = output_high
+ * else:
+ *   output = round((x - input_low) / (input_high - input_low) * (levels-1)) / (levels-1) * (output_high - output_low) + output_low
+ *
+ * expand brackets into round:
+ * round(x * (levels-1) / (input_high - input_low) - input_low * (levels-1) / (input_high - input_low))
+ * div on (levels-1) and mult on (output_high - output_low) => mult on (output_high - output_low) / (levels-1)
+ *
+ *  =>
+ * round(x * (levels-1) / (input_high - input_low) - input_low * (levels-1) / (input_high - input_low)) * (output_high - output_low) / (levels-1) + output_low
+*/
+ngraph::pass::FakeQuantizeDecomposition::FakeQuantizeDecomposition() {
+    MATCHER_SCOPE(FakeQuantizeDecomposition);
+    auto fake_quantize = ngraph::pattern::wrap_type<ngraph::opset1::FakeQuantize>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto &pattern_to_output = m.get_pattern_value_map();
+        const auto fake_quantize_node = std::dynamic_pointer_cast<ngraph::opset1::FakeQuantize>(pattern_to_output.at(fake_quantize).get_node_shared_ptr());
+
+        if (fake_quantize_node == nullptr || !isNeedToDecomposFQ(fake_quantize_node)) {
+            return false;
+        }
+
+        Output<Node> data{fake_quantize_node->input_value(0)};
+        const Output<Node> input_low{fake_quantize_node->input_value(1)};
+        const Output<Node> input_high{fake_quantize_node->input_value(2)};
+        const Output<Node> output_low{fake_quantize_node->input_value(3)};
+        const Output<Node> output_high{fake_quantize_node->input_value(4)};
+        auto input_type = data.get_element_type();
+
+        ngraph::NodeVector decomp_ops;
+        if (input_type != input_low.get_element_type()) {
+            input_type = input_low.get_element_type();
+            data = std::make_shared<ngraph::opset1::Convert>(data, input_type);
+            decomp_ops.push_back(data.get_node_shared_ptr());
+        }
+
+        // if we set input_low or input_high in formula we got output = output_low and = output_high respectively
+        // so we just clamp x
+        const auto max = std::make_shared<ngraph::opset1::Maximum>(data, input_low);
+        const auto min = std::make_shared<ngraph::opset1::Minimum>(max, input_high);
```

Comment:
Is it valid for case "input_high < input_low"?

Reply:
Unsupported case
Added into validation 'range' nodes function